### PR TITLE
OAK-11221: replaced guava evicting queue with LinkedhashMap

### PR DESCRIPTION
**Pull request by rishabhdaim**

### CODE REVIEW

**Review by mbaedke — APPROVED** [review]

File: oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java [hunk 1]
```diff
@@ -275,13 +281,13 @@ public String pickRandomPath(){
             if (assets != null){
                 return assets.get(random.nextInt(assets.size()));
             }
-            return buffer.peek();
+            return buffer.keySet().stream().findFirst().orElse(null);
```
Comment (reschke) [inline oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java:284]: I'd avoid this change in the context of this ticket.

File: oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java [hunk 2]
```diff
@@ -275,13 +281,13 @@ public String pickRandomPath(){
...
         }
 
         public void addAssetPath(String path) {
-            buffer.add(path);
+            buffer.put(path, Boolean.TRUE);
```
Comment (reschke) [inline oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java:288]: same here

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/report/LoggingReporter.java [hunk 3]
```diff
@@ -77,10 +84,10 @@ protected void reportPeriodicProperty(final long count, @NotNull final Reporting
...
         String path = nodeState.getPath();
-        if (lastPaths.contains(path)) {
+        if (lastPaths.containsKey(path)) {
             return true;
         } else {
-            lastPaths.add(path);
+            lastPaths.put(path, Boolean.TRUE);
             return false;
         }
     }
```
Comment (reschke) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/report/LoggingReporter.java:93]: same here

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/secondary/SecondaryStoreCache.java [hunk 4]
```diff
@@ -181,8 +187,8 @@ private AbstractDocumentNodeState findMatchingRoot(RevisionVector rr) {
     public void contentChanged(@NotNull AbstractDocumentNodeState root) {
         synchronized (queue){
             //TODO Possibly can be improved
-            queue.add(root);
-            previousRoots = queue.toArray(EMPTY);
+            queue.put(root, Boolean.TRUE);
+            previousRoots = queue.keySet().toArray(EMPTY);
```
Comment (reschke) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/secondary/SecondaryStoreCache.java:191]: same here

**Review by reschke — CHANGES_REQUESTED** [review]
-please separate unrelated changes-

File: oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java (hunk 1 above)
Comment (rishabhdaim) [reply oak-benchmarks/src/main/java/org/apache/jackrabbit/oak/benchmark/BundlingNodeTest.java:284]: this is related to change.

We have updated the type of buffer to LinkedHashMap from EvictingQueue, so we need to change the above.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/report/LoggingReporter.java (hunk 3 above)
Comment (rishabhdaim) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/report/LoggingReporter.java:93]: same as above.

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/secondary/SecondaryStoreCache.java (hunk 4 above)
Comment (rishabhdaim) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/secondary/SecondaryStoreCache.java:191]: same as above.

**Review by reschke — APPROVED** [review]

**Comment by rishabhdaim** [comment]
thanks @reschke for review.

**Pull request merged**